### PR TITLE
Adding expose flag for ingress

### PR DESCRIPTION
**Pull request by RakhithaRR**

### CODE REVIEW

File: scenarios/scenario-1/wso2-apim.yaml [hunk 1]
```diff
@@ -4,6 +4,7 @@ metadata:
   name: cluster-1
 spec:
   pattern: Pattern-1
+  expose: ""
```
Comment (pubudu538) [inline scenarios/scenario-1/wso2-apim.yaml]: Why this is defined here? It is not required and it is an optional field.

**Review by pubudu538 — CHANGES_REQUESTED** [review]

**Review by pubudu538 — APPROVED** [review]

File: artifacts/install/operator-artifacts/6-controller.yaml [hunk 2]
```diff
@@ -15,6 +15,6 @@ spec:
     spec:
       containers:
         - name: wso2am-controller
-          image: pubudu/k8s-wso2am-operator:0.8.1
+          image: rakhitharr/wso2am-controller:v17
           imagePullPolicy: Always
       serviceAccountName: wso2am-pattern-1-svc-account
```
Comment (praminda) [inline artifacts/install/operator-artifacts/6-controller.yaml:20]: Add a newline at the end of the file

File: pkg/controller/controller.go [hunk 3]
```diff
@@ -560,37 +560,58 @@ func (c *Controller) syncHandler(key string) error {
...
-		apimingressname := "wso2-am-p1-ingress"
-		amingress, err := c.ingressLister.Ingresses(apimanager.Namespace).Get(apimingressname)
-		// If the resource doesn't exist, we'll create it
-		if errors.IsNotFound(err) {
-			amingress, err = c.kubeclientset.ExtensionsV1beta1().Ingresses(apimanager.Namespace).Create(pattern1.ApimIngress(apimanager))
-			if err != nil {
-				return err
+		if apimanager.Spec.Expose == "Ingress" {
+			// Get apim instance 1 service name using hardcoded value
+			apimingressname := "wso2-am-p1-ingress"
```
Comment (praminda) [inline pkg/controller/controller.go:565]: shall we make var names camel case?

File: artifacts/install/operator-artifacts/6-controller.yaml (hunk 2 above)
Comment (mushthaq33) [reply artifacts/install/operator-artifacts/6-controller.yaml:20]: Do we normally use a user specific location?

File: scenarios/scenario-1/wso2-apim.yaml (hunk 1 above)
Comment (RakhithaRR) [reply scenarios/scenario-1/wso2-apim.yaml]: Fixed with c4be73d

File: artifacts/install/operator-artifacts/6-controller.yaml (hunk 2 above)
Comment (RakhithaRR) [reply artifacts/install/operator-artifacts/6-controller.yaml:20]: New line is added in the latest code
This image was used for pre-release testing. It has been updated to wso2 images in latest code.

File: pkg/controller/controller.go (hunk 3 above)
Comment (RakhithaRR) [reply pkg/controller/controller.go:565]: Fixed with https://github.com/wso2/k8s-wso2am-operator/pull/77

**Pull request merged**